#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_empty-lhs-sparse.dpatch by  <rafael@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: New patch generated from octave2.9 2.9.9-4 diff.gz

@DPATCH@

--- octave2.9-2.9.9.orig/doc/conf.texi
+++ octave2.9-2.9.9/doc/conf.texi
@@ -3,7 +3,7 @@
 @c For copying conditions, see the file gpl.texi.
 
 @set OCTAVE_MANUAL
-@set OCTAVEHOME /usr/local
+@set OCTAVEHOME /usr
 @set VERSION 2.9.9
 @set HAVE_COLAMD
 @set HAVE_CHOLMOD
--- octave2.9-2.9.9.orig/liboctave/Sparse.cc
+++ octave2.9-2.9.9/liboctave/Sparse.cc
@@ -2387,6 +2387,8 @@
     {
       (*current_liboctave_error_handler)
         ("A(I, J) = X: can only have 1 or 2 indexes for sparse matrices");
+
+      lhs.clear_index ();
       return 0;
     }
 
@@ -2482,8 +2484,9 @@
 			      octave_idx_type iii = 0;
 			      octave_idx_type ii = idx_i.elem (iii);
 			      octave_idx_type ppp = 0;
-			      octave_idx_type ppi = lhs.cidx(j+1) - 
-				lhs.cidx(j); 
+			      octave_idx_type ppi = (j >= lhs_nc ? 0 : 
+						     lhs.cidx(j+1) - 
+						     lhs.cidx(j));
 			      octave_idx_type pp = (ppp < ppi ? 
 						    lhs.ridx(lhs.cidx(j)+ppp) :
 						    new_nr);
@@ -2634,8 +2637,9 @@
 			      octave_idx_type iii = 0;
 			      octave_idx_type ii = idx_i.elem (iii);
 			      octave_idx_type ppp = 0;
-			      octave_idx_type ppi = lhs.cidx(j+1) -
-				lhs.cidx(j);
+			      octave_idx_type ppi = (j >= lhs_nc ? 0 : 
+						     lhs.cidx(j+1) - 
+						     lhs.cidx(j));
 			      octave_idx_type pp = (ppp < ppi ? 
 						    lhs.ridx(lhs.cidx(j)+ppp) :
 						    new_nr);
--- octave2.9-2.9.9.orig/src/__gnuplot_raw__.cc
+++ octave2.9-2.9.9/src/__gnuplot_raw__.cc
@@ -3380,9 +3380,11 @@
   octave_value_list args;
 
   args(0) = "off";
-
   feval ("hold", args);
 
+  args.resize (0);
+  feval ("__clear_plot__", args);
+
   return octave_value_list ();
 }
 
